log/stdlib_test.go

```diff
@@ -31,13 +31,13 @@ func TestStdlibAdapterUsage(t *testing.T) {
 	time := now.Format("15:04:05")
 
 	for flag, want := range map[int]string{
-		0:                                      "msg=hello\n",
-		log.Ldate:                              "ts=" + date + " msg=hello\n",
-		log.Ltime:                              "ts=" + time + " msg=hello\n",
-		log.Ldate | log.Ltime:                  "ts=\"" + date + " " + time + "\" msg=hello\n",
-		log.Lshortfile:                         "caller=stdlib_test.go:44 msg=hello\n",
-		log.Lshortfile | log.Ldate:             "ts=" + date + " caller=stdlib_test.go:44 msg=hello\n",
-		log.Lshortfile | log.Ldate | log.Ltime: "ts=\"" + date + " " + time + "\" caller=stdlib_test.go:44 msg=hello\n",
+		0:                                      "msg=\"hello\\n\"\n",
+		log.Ldate:                              "ts=" + date + " msg=\"hello\\n\"\n",
+		log.Ltime:                              "ts=" + time + " msg=\"hello\\n\"\n",
+		log.Ldate | log.Ltime:                  "ts=\"" + date + " " + time + "\" msg=\"hello\\n\"\n",
+		log.Lshortfile:                         "caller=stdlib_test.go:44 msg=\"hello\\n\"\n",
+		log.Lshortfile | log.Ldate:             "ts=" + date + " caller=stdlib_test.go:44 msg=\"hello\\n\"\n",
+		log.Lshortfile | log.Ldate | log.Ltime: "ts=\"" + date + " " + time + "\" caller=stdlib_test.go:44 msg=\"hello\\n\"\n",
```

Comment:
This test case will not change if we strip the trailing newline heres:
```diff
diff --git a/log/stdlib.go b/log/stdlib.go
index 2d2e9c0..1e0e19f 100644
--- a/log/stdlib.go
+++ b/log/stdlib.go
@@ -145,7 +145,7 @@ func subexps(line []byte) map[string]string {
        }
        result := map[string]string{}
        for i, name := range logRegexp.SubexpNames() {
-               result[name] = string(m[i])
+               result[name] = strings.TrimSuffix(string(m[i]), "\n")
        }
        return result
 }
```

Reply:
Thanks, I was thinking about this, but wasn't sure if that's what we wanted.